src/Runner.Common/HostContext.cs

```diff
@@ -83,6 +83,7 @@ public HostContext(string hostType, string logFile = null)
             _loadContext.Unloading += LoadContext_Unloading;
 
             this.SecretMasker.AddValueEncoder(ValueEncoders.Base64StringEscape);
+            this.SecretMasker.AddValueEncoder(ValueEncoders.Base64StringEscapeTrimmed);
```

Comment:
I made the decision to treat this as a separate encoder, which will slightly affect runtime performance. 

We want to ensure as a contract to the user that if they register a secret, we will mask it and various encodings of it (base64, JSON, XML). We also want to best effort mask what we can if the secret is base64 encoded as a part of a larger string. We will leak a few bits a bit of information, but the majority of the secret will be masked.